Update docs

holmanb opened the pull request:
## Additional Context

Documents #5489, sprinkles in some more links to connect related pages, and moves the first boot determination docs into a dedicated page.

## Merge type

- [x] Squash merge using "Proposed Commit Message"
- [ ] Rebase and merge unique commits. Requires commit messages per-commit each referencing the pull request number (#<PR_NUM>)

a-dubs commented on doc/rtd/explanation/boot.rst:
```diff
@@ -89,9 +88,8 @@ including persistent device naming with old MAC addresses.
-as DHCP hooks or broadcast of an old hostname. It would also put the system
-in an odd state to recover from, as it may then have to restart network
-devices.
+broadcast of an old hostname. It would also put the system in an odd state to
```
Missing an "as" at the beginning

a-dubs reviewed (changes requested):
Looks great! One small typo. Thanks for updating docs! :blue_heart:

holmanb commented on doc/rtd/explanation/analyze.rst:
```diff
@@ -3,15 +3,18 @@
 - :command:`dump`
 - :command:`boot`
 
+The analyze subcommand works by parsing the cloud-init log file for timestamps
```
This subcommand fails to capture interpreter startup time and library load time because of this fact, so at least advertise how it works so that users can draw their own conclusions.

TheRealFalcon commented on cloudinit/cmd/main.py:
```diff
@@ -1033,7 +1037,11 @@ def main(sysv_args=None):
+        help=(
+            "Manually run a single module. Useful for "
+            "testing during development."
+        ),
```
Should we add DEPRECATED to `--force` and `--debug` when called from the CLI too?

holmanb commented on doc/rtd/explanation/instancedata.rst:
```diff
@@ -65,7 +65,6 @@ Using ``instance-data``
 
 ``instance-data`` can be used in:
 
-* :ref:`User data scripts<user_data_script>`.
```
This was redundant, since user data scripts are documented on the user data formats page.

holmanb commented on doc/rtd/explanation/kernel-command-line.rst:
```diff
@@ -2,18 +2,12 @@ Kernel command line
 
 Providing configuration data via the kernel command line is somewhat of a last
-resort, since this method only supports
-:ref:`cloud config<user_data_formats-cloud_config>` starting with
```
This piece of information doesn't apply to datasource discovery override, and is already mentioned in the section below that it is relevant to. Drop it.

TheRealFalcon commented on doc/rtd/explanation/first_boot.rst:
```diff
@@ -0,0 +1,91 @@
+.. _First_boot_determination:
```
Did you change any of the text on this page? Just wondering if I should be paying attention to anything specific.

holmanb replied:
Nope!

holmanb commented:
> Looks great! One small typo. Thanks for updating docs! 💙

Thanks for the review @a-dubs. I just added some more content and fixed the issue you pointed out!

TheRealFalcon commented on doc/rtd/explanation/instancedata.rst:
```diff
@@ -65,7 +65,6 @@ Using ``instance-data``
 
 ``instance-data`` can be used in:
 
-* :ref:`User data scripts<user_data_script>`.
```
Why the removal? [It's a valid usage](https://cloudinit.readthedocs.io/en/latest/explanation/format.html#example-user-data-script)

holmanb replied:
This was redundant, since user data scripts are documented on the user data formats page.

TheRealFalcon replied:
Gotcha. As-is, it's a little weird though. The text says `Cloud-config` but links to the user data page (not your doing). We could update the text of the link to say 'user data' rather than 'cloud-config',  you can only use instance data in cloud-config and user scripts, so I'm not sure we should be saying it can be used with user data when there are caveats to that.

I think we should either specify cloud-config and user data scripts independently and be ok with the redundancy, or we should link to the jinja format and say any user data format supported by jinja.

holmanb replied:
@TheRealFalcon good point. I just pushed an update, let me know what you think.

TheRealFalcon commented on doc/rtd/howto/rerun_cloud_init.rst:
```diff
@@ -64,33 +64,10 @@ a result.
 
 .. _partially_rerun_cloud_init:
 
-How to partially re-run cloud-init
```
I think this information is still very useful for debugging. I've typed this out manually to various different people at least a dozen times. Add a million warnings around it if you want or change it to `cloud-init --all-stages`, but I think we should keep it.

holmanb replied:
If we're calling it deprecated and planning on eventual removal, I'd rather not keep the old commands around.

> Add a million warnings around it if you want or change it to cloud-init --all-stages, but I think we should keep it.

I'll go that route.

TheRealFalcon commented on doc/rtd/reference/breaking_changes.rst:
```diff
@@ -11,6 +11,45 @@ releases.
+---------------------------
+
+As a performance optimization, cloud-init no longer runs as four seperate
+Python interpretters. Instead, it launches a single process and then
```
```suggestion
Python invocations. Instead, it launches a single process and then
```

`interpreters` makes me think we have independent interpreters installed or something.

TheRealFalcon reviewed (changes requested):
I left a few comments inline, but nothing major.

holmanb commented:
@TheRealFalcon thanks for your review. I think that I've addressed all of your comments.

TheRealFalcon commented on doc/rtd/explanation/instancedata.rst:
```diff
@@ -63,9 +63,10 @@ provided to this instance. Non-root users referencing ``userdata`` or
 
-* :ref:`Cloud-config data<user_data_formats>`.
+* :ref:`User data scripts<user_data_script>`.
+* :ref:`User data<user_data_formats-cloud_config>`.
```
`Cloud-config` instead of `User_data`?

holmanb commented:
@TheRealFalcon ready for re-review, thanks!

TheRealFalcon reviewed (approved):
LGTM!

holmanb commented:
I just rebased the fixup commits and updated commit messages with PR number.
